ui/round/src/view/user.ts

```diff
@@ -66,10 +68,38 @@ export function userHtml(ctrl: RoundController, player: Player, position: Positi
   );
 }
 
-const signalBars = (signal: number) => {
-  const bars = [];
+const bars = (signal: number) => {
+  const bars: VNode[] = [];
   for (let i = 1; i <= 4; i++) bars.push(h(i <= signal ? 'i' : 'i.off'));
-  return h('signal.q' + signal, bars);
+  return bars;
+};
+
+const playerLag = () => {
+  const ping = Math.round(wsAverageLag());
+  return !ping ? 0 : ping < 150 ? 4 : ping < 300 ? 3 : ping < 500 ? 2 : 1;
```

Comment:
I'd refactor all the ternaries to a switch statement. This is a bit hard to read.

Reply:
switch is heavier and will consume more bandwidth
in addition, this ternary is already used in other places

Reply:
Huh? A switch will consume more bandwidth?

Reply:
Each additional character added to a file will add more bytes that will have to be used on the user's bandwidth.
In other words, an unnecessary switch will cause the user to consume more and the page to load more slowly.

Ternary is used here too:
https://github.com/lichess-org/lila/blob/0d36d7615e5083b95bb31a246d24d4ed32ff1f21/ui/dasher/src/ping.ts#L48

Reply:
Yeah, that reasoning about bandwidth is a bit of a stretch. Optimizing for a few extra characters in a minified TypeScript/JavaScript file isn’t going to make any noticeable difference in real-world performance. Readability should take priority in cases like this, and a deeply nested ternary can be harder to parse than a switch or if-else.

Also, in this specific case, I’m 100% sure that this level of optimization is unnecessary. The actual impact on performance is nonexistent, but the impact on code clarity is very real.

I'm not going to dwell on it though, since I don't know the codebase well 😅.

Reply:
you mean if/else (switch cases cannot be dynamic). i agree the bandwidth isn't an issue but the readability of ternaries vs if/else chains can also be debated, and our practice is to defer to contributor's choice when there's a bit of grey area.

Reply:
My 2 cents: I like the ternary operator, not for optimization or bandwidth, but because of readability.

`if` and `switch` are statements and require mutating a value. The ternary operator is an expression that only returns a value. These almost always make more sense to me.